src/serde/ser.rs

```diff
@@ -0,0 +1,991 @@
+    }
+
+    /// Serialize all integer types using the `Integer` intermediary type.
+    fn serialize_u8(self, v: u8) -> Result<Self::Ok, Self::Error> {
+        self.writer.write_int(&Int::from(v))
+    }
+
+    /// Serialize all integer types using the `Integer` intermediary type.
+    fn serialize_i16(self, v: i16) -> Result<Self::Ok, Self::Error> {
+        self.writer.write_int(&Int::from(v))
+    }
+
+    /// Serialize all integer types using the `Integer` intermediary type.
+    fn serialize_u16(self, v: u16) -> Result<Self::Ok, Self::Error> {
+        self.writer.write_int(&Int::from(v))
+    }
+
+    /// Serialize all integer types using the `Integer` intermediary type.
+    fn serialize_i32(self, v: i32) -> Result<Self::Ok, Self::Error> {
+        self.writer.write_int(&Int::from(v))
+    }
+
+    /// Serialize all integer types using the `Integer` intermediary type.
+    fn serialize_u32(self, v: u32) -> Result<Self::Ok, Self::Error> {
+        self.writer.write_int(&Int::from(v))
+    }
+
+    /// Serialize all integer types using the `Integer` intermediary type.
+    fn serialize_i64(self, v: i64) -> Result<Self::Ok, Self::Error> {
+        self.writer.write_int(&Int::from(v))
+    }
+
+    /// Serialize all integer types using the `Integer` intermediary type.
+    fn serialize_u64(self, v: u64) -> Result<Self::Ok, Self::Error> {
+        self.writer.write_int(&Int::from(v))
+    }
+
+    fn serialize_f32(self, v: f32) -> Result<Self::Ok, Self::Error> {
+        self.writer.write_f32(v)
+    }
+
+    fn serialize_f64(self, v: f64) -> Result<Self::Ok, Self::Error> {
+        self.writer.write_f64(v)
+    }
+
+    fn serialize_char(self, v: char) -> Result<Self::Ok, Self::Error> {
+        self.writer.write_string(v.to_string())
+    }
+
+    fn serialize_str(self, v: &str) -> Result<Self::Ok, Self::Error> {
+        self.writer.write_string(v)
+    }
+
+    fn serialize_bytes(self, v: &[u8]) -> Result<Self::Ok, Self::Error> {
+        self.writer.write_blob(v)
+    }
+
+    fn serialize_none(self) -> Result<Self::Ok, Self::Error> {
+        self.writer.write_null(IonType::Null)
+    }
+
+    fn serialize_some<T: ?Sized>(self, value: &T) -> Result<Self::Ok, Self::Error>
+    where
+        T: Serialize,
+    {
+        value.serialize(self)
+    }
+
+    fn serialize_unit(self) -> Result<Self::Ok, Self::Error> {
+        self.serialize_none()
+    }
+
+    fn serialize_unit_struct(self, _name: &'static str) -> Result<Self::Ok, Self::Error> {
+        self.serialize_unit()
+    }
+
+    fn serialize_unit_variant(
+        self,
+        _name: &'static str,
+        _variant_index: u32,
+        variant: &'static str,
+    ) -> Result<Self::Ok, Self::Error> {
+        self.serialize_str(variant)
+    }
+
+    fn serialize_newtype_struct<T: ?Sized>(
+        self,
+        name: &'static str,
+        value: &T,
+    ) -> Result<Self::Ok, Self::Error>
+    where
+        T: Serialize,
+    {
+        if name == "$__ion_rs_timestamp__" {
```

Comment:
Can you add a `const` for `"$__ion_rs_timestamp__"` and `"$__ion_rs_decimal__"`? (See [this SO answer](https://softwareengineering.stackexchange.com/a/365344) if you want more information about why magic strings are generally not a good idea.)